marlowe-playground-client/src/HaskellEditor/Types.purs

```diff
@@ -36,28 +38,37 @@ instance actionIsEvent :: IsEvent Action where
   toEvent SendResultToBlockly = Just $ defaultEvent "SendResultToBlockly"
 
 type State
-  = { activeDemo :: String
-    , keybindings :: KeyBindings
+  = { keybindings :: KeyBindings
     , compilationResult :: WebData (Either InterpreterError (InterpreterResult String))
     , showBottomPanel :: Boolean
     }
 
-_activeHaskellDemo :: Lens' State String
-_activeHaskellDemo = prop (SProxy :: SProxy "activeDemo")
-
 _haskellEditorKeybindings :: Lens' State KeyBindings
 _haskellEditorKeybindings = prop (SProxy :: SProxy "keybindings")
 
 _compilationResult :: Lens' State (WebData (Either InterpreterError (InterpreterResult String)))
 _compilationResult = prop (SProxy :: SProxy "compilationResult")
 
+--- Language.Haskell.Interpreter is missing this ---
+_result :: forall s a. Lens' { result :: a | s } a
+_result = prop (SProxy :: SProxy "result")
+
+_Pretty :: Getter' String String
+_Pretty = to f
+  where
+  f ugly = case parseContract ugly of
+    Right contract -> (show <<< pretty) contract
+    Left _ -> ugly
+
+_ContractString :: forall r. Monoid r => Fold' r State String
+_ContractString = _compilationResult <<< _Success <<< _Right <<< _InterpreterResult <<< _result <<< _Pretty
```

Comment:
Won't this replace any unparsable contract with the empty string? 🤔

Reply:
no, this will replace any unparseable contract with the original version or alternatively, prettify it if it can

Reply:
specifically it the interpreter result could be either an error string or a contract and we want to display them in the same way but prettify the contract if we can. Maybe the name is wrong

Reply:
Ah, yep. I've been misreading the Getter and interpreting it as a Prism. Ignore me. 👍